File: test/aidbox_sdk/snapshots/hl7-fhir-r4-core/Communication.cs
using Aidbox.FHIR.Base;
using Aidbox.FHIR.Utils;

namespace Aidbox.FHIR.Resource;

public class Communication : DomainResource, IResource
{
    public Base.CodeableConcept[]? Category { get; set; }
    public string? Received { get; set; }
    public string[]? InstantiatesCanonical { get; set; }
    public Communication_Payload[]? Payload { get; set; }
    public string[]? InstantiatesUri { get; set; }
    public Base.ResourceReference? Encounter { get; set; }
    public Base.CodeableConcept[]? Medium { get; set; }
    public Base.ResourceReference[]? Recipient { get; set; }
    public Base.CodeableConcept[]? ReasonCode { get; set; }
    public Base.CodeableConcept? StatusReason { get; set; }
    public Base.CodeableConcept? Topic { get; set; }
    public string? Sent { get; set; }
    public Base.Annotation[]? Note { get; set; }
    public string? Priority { get; set; }
    public required string Status { get; set; }
    public Base.ResourceReference? Sender { get; set; }
    public Base.Identifier[]? Identifier { get; set; }
    public Base.ResourceReference[]? InResponseTo { get; set; }
    public Base.ResourceReference[]? BasedOn { get; set; }
    public Base.ResourceReference[]? PartOf { get; set; }
    public Base.ResourceReference? Subject { get; set; }
    public Base.ResourceReference[]? About { get; set; }
    public Base.ResourceReference[]? ReasonReference { get; set; }

    public class Communication_Payload : BackboneElement
    {
        public object? Content    
        {
            get
            {
                if (ContentString is not null)
                {
                    return ContentString;
                }
        
                if (ContentReference is not null)
                {
                    return ContentReference;
                }
        
                if (ContentAttachment is not null)
                {
                    return ContentAttachment;
                }
        
                return null;
            }
        
            set
            {
                if (value?.GetType() == typeof(string))
                {
                    ContentString = (string)value;return;
                }
        
                if (value?.GetType() == typeof(Base.ResourceReference))
                {
                    ContentReference = (Base.ResourceReference)value;return;
                }
        
                if (value?.GetType() == typeof(Base.Attachment))
                {
                    ContentAttachment = (Base.Attachment)value;return;
                }
        
                throw new ArgumentException("Invalid type provided");
            }
        }
        public string? ContentString { get; set; }
        public Base.ResourceReference? ContentReference { get; set; }
        public Base.Attachment? ContentAttachment { get; set; }
    }
}

File: test/aidbox_sdk/snapshots/python/hl7-fhir-r4-core/Communication.py
from typing import Optional, List
from pydantic import *
from ..base.Annotation import Annotation
from ..base.Attachment import Attachment
from ..base.CodeableConcept import CodeableConcept
from ..base.DomainResource import DomainResource
from ..base.Reference import Reference
from ..base.Identifier import Identifier
from ..base.BackboneElement import BackboneElement

class Communication_Payload(BackboneElement):
    content_string: Optional[str] = None
    content_reference: Optional[Reference] = None
    content_attachment: Optional[Attachment] = None

class Communication(DomainResource):
    category: Optional[List[CodeableConcept]] = None
    received: Optional[str] = None
    instantiates_canonical: Optional[List[str]] = None
    payload: Optional[List[Communication_Payload]] = None
    instantiates_uri: Optional[List[str]] = None
    encounter: Optional[Reference] = None
    medium: Optional[List[CodeableConcept]] = None
    recipient: Optional[List[Reference]] = None
    reason_code: Optional[List[CodeableConcept]] = None
    status_reason: Optional[CodeableConcept] = None
    topic: Optional[CodeableConcept] = None
    sent: Optional[str] = None
    note: Optional[List[Annotation]] = None
    priority: Optional[str] = None
    status: str
    sender: Optional[Reference] = None
    identifier: Optional[List[Identifier]] = None
    in_response_to: Optional[List[Reference]] = None
    based_on: Optional[List[Reference]] = None
    part_of: Optional[List[Reference]] = None
    subject: Optional[Reference] = None
    about: Optional[List[Reference]] = None
    reason_reference: Optional[List[Reference]] = None